### DISCUSSION TITLE
Limit by Date is confusing

### DISCUSSION THREAD

**Discussion by thany** [MalloyDelacroix/DownloaderForReddit#265] [Q&A] [answered]
Let me just sum up the questions that pop up in my head:

1. `Limit by Date` - Limit how exactly? Does it not go further back that what I set there, or the reverse?
2. `Last Post Date:` - What does the colon mean / what is it pointing to? What does this label mean? It feels like "Don't download anything older than the last post date", which suggests, at least theoretically, that it will download 1 post - because all other posts are older than the last post date.
3. `Update custom limit` - Update by what?

In the previous major version, I've always just set the custom date limit to the day that I last performed a download. If that's now done automatically, `Update custom limit` should probably be a checkbox saying `Automatially limit by last download date`. Checking this should then disable the date field (and therefor the checkbox should be above the date field), because filling it out is done automatically by the program.

Looking at it from a different angle:

Perhaps change it to `Maximum post age in days` which is far easier to understand, than to deal with fixed dates. This maximum age could also optionally be set automatically to however many days ago it was since the last successful download.

Dealing with dates is a lot harder in my brain, than it is to think about "X days ago" or similar.

**Comment by MalloyDelacroix** [comment, marked as answer]
1.)  The limit by date checkbox enacts the date limit controls below it.  If it is not checked, date of post is not considered when extracting posts from reddit.  But date limits do mean that no post older than that date will be extracted.

2.)  Last post date means that it will not extract any post older than the newest post that has already been extracted for that user/subreddit.  The date shown to the right of the colon is the newest post date that has been extracted for that user/subreddit.  It is updated anytime a post is downloaded.  

3.)  Custom date limit lets you set your own date limit that is not influenced by the post date, at first.  If you select "keep custom limit", that date is not changed after performing a download.  If you select "update custom limit", it will use your custom date for the next download run, then it will revert to keeping track of the last extracted post date, and use it as the limit instead.  This is intended for one-off downloads where you may want to go further back for the next download, but only the one time.

It should be noted about this section: when the extractor is working on extracting posts for the user/subreddit it is working on, it will stop when the date limit is reached.  If there is no date limit or the date limit is in the distant past, it will keep looking until it hits the post limit.  So not setting a date limit or setting a distant limit will increase run time.

I like the idea of displaying the number of days instead of a hard date.  That most likely would improve the understanding of this section.  Maybe a hybrid approach where it switches to a date display after so many days has elapsed.  After a certain number of days, a day count almost loses meaning.

It does automatically update the date to the date of the newest post that has been extracted for the user/subreddit.  I can see that some of this UI is unclear.  Some parts do have tool-tips that should help explain.  Ultimately though, it is hard to ride the line of displaying enough information on the UI and it all just becoming a cluttered mess.  I'm definitely not a UI designer, so I do struggle with this.
